docs/cluster_install.md

```diff
@@ -0,0 +1,314 @@
+    ```
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+    add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+    apt-get install docker-ce docker-ce-cli containerd.io
+    ```
+
+6. Install jq (if not already installed):
+    ```
+    apt-get install jq
+    ```
+
+7. Define this registry to docker as an insecure registry:
+
+    a. Run the following to define an insecure registry route using the value of the $REGISTRY_ENDPOINT environment variable obtained in the last step and append it to the /etc/docker/daemon.json file.
+
+    ```
+    echo "{
+        \"insecure-registries\": [ \"$REGISTRY_ENDPOINT\" ]
+    }" >> /etc/docker/daemon.json
+    ```
+
+    b. (optional) Verify that docker is on your machine:
+    ```
+    curl -fsSL get.docker.com | sh
+    ```
+
+    c. Restart docker to pick up the change:
+    ```
+    systemctl restart docker
+    ```
+
+## Install and configure a microk8s edge cluster
+**Note**: If you already have a microk8s cluster installed, skip to the next section: Installing the Cluster Agent.
+
+This content provides a summary of how to install microk8s, a lightweight and small Kubernetes cluster, on Ubuntu 18.04. (For more information, see the microk8s documentation.)
+
+**Note**: This type of edge cluster is meant for development and test because a single worker node Kubernetes cluster does not provide scalability or high availability.
+
+1. Install microk8s:
+    ```
+    sudo snap install microk8s --classic --channel=stable
+    ```
+
+2. If you are not running as root, add your user to the microk8s group:
+    ```
+    sudo usermod -a -G microk8s $USER
+    sudo chown -f -R $USER ~/.kube
+    su - $USER
+    ```
+
+3. Enable dns and storage modules in microk8s:
+    ```
+    microk8s.enable dns
+    microk8s.enable storage
+    ```
+    **Note**: Microk8s uses 8.8.8.8 and 8.8.4.4 as upstream name servers by default. If these name servers cannot resolve the management hub hostname, you must change the name servers that microk8s is using:
+
+    a. Retrieve the list of upstream name servers in `/etc/resolv.conf` or `/run/system/resolve/resolv.conf`
+
+    b. Edit coredns configmap in the kube-system namespace. Set the upstream nameservers in the forward section
+    ```
+    microk8s.kubectl edit -n kube-system cm/coredns
+    ```
+
+4. Check the status:
+    ```
+    microk8s.status --wait-ready
+    ```
+
+5. The microk8s kubectl command is called microk8s.kubectl to prevent conflicts with an already install kubectl command. Assuming that kubectl is not installed, add this alias for microk8s.kubectl:
+    ```
+    echo 'alias kubectl=microk8s.kubectl' >> ~/.bash_aliases
+    source ~/.bash_aliases
+    ```
+
+6. Enable the container registry and configure docker to tolerate the insecure registry:
+
+    a. Enable the container registry
+    ```
+    microk8s.enable registry
+    export REGISTRY_ENDPOINT=localhost:32000
```

Comment:
Can you check this PR: https://github.com/open-horizon/open-horizon.github.io/pull/272/files? There is another env introduced `REGISTRY_IP_ENDPOINT` because localhost:32000 doesn't if push image from inside of agent pod

Reply:
So you will need to add `export REGISTRY_IP_ENDPOINT=$(kubectl get service registry -n container-registry | grep registry | awk '{print $3;}'):5000` after line 204